src/Integration/ModulesIntegration.php

```diff
@@ -24,26 +24,19 @@ final class ModulesIntegration implements IntegrationInterface
     /**
      * {@inheritdoc}
      */
-    public function setupOnce(): void
+    public function bindToHub(HubInterface $hub): IntegrationInterface
     {
-        Scope::addGlobalEventProcessor(function (Event $event) {
-            $self = Hub::getCurrent()->getIntegration(self::class);
+        Scope::addGlobalEventProcessor([$this, 'applyToEvent']);
 
-            if ($self instanceof self) {
-                self::applyToEvent($self, $event);
-            }
-
-            return $event;
-        });
+        return $this;
     }
 
     /**
      * Applies the information gathered by this integration to the event.
      *
-     * @param self  $self  The instance of this integration
      * @param Event $event The event that will be enriched with the modules
      */
-    public static function applyToEvent(self $self, Event $event): void
+    public static function applyToEvent(Event $event): void
```

Comment:
Since this method should not be called directly from the outside, make it `private`